apps/bbsync/tests/test_query.py

```diff
@@ -1252,3 +1254,58 @@ def test_generate_nist_cvss_validation(
             assert {"name": "nist_cvss_validation", "status": flag_value} in flags
         else:
             assert flags == []
+
+
+class TestGenerateFields:
+    def test_generate_fixed_in_without_meta_attr(self):
+        """
+        Tests that package_versions are not converted to fixed_in bz field when
+        it's not clear where to use dash or space as separator.
+        """
+        flaw = FlawFactory(source=FlawSource.CUSTOMER)
+        # fixed_in NOT in meta_attr!
+        AffectFactory(flaw=flaw)
+
+        pkg = PackageFactory(package="foobar", flaw=flaw)
+        PackageVerFactory(version="1.2.3.4", package=pkg)
+
+        fbqb = FlawBugzillaQueryBuilder(flaw)
+
+        # Test that nothing is generated when fixed_in is not present in meta_attr.
+        # This is so that existing historical fixed_in is not destroyed (there's
+        # a lot of non-package-version strings with dashes in old flaws).
+
+        assert "cf_fixed_in" not in fbqb.query
+
+    def test_generate_fixed_in(self):
+        """
+        Tests that package_versions are correctly converted to fixed_in bz field.
+        """
+        flaw = FlawFactory(source=FlawSource.CUSTOMER)
+        flaw.meta_attr[
+            "fixed_in"
+        ] = "bazfoo-2.3.4.5, something 4.5, foobar 1.2.3.4, foobar-2.3.4.5"
+        AffectFactory(flaw=flaw)
+
+        pkg_a = PackageFactory(package="foobar", flaw=flaw)
+        pkg_b = PackageFactory(package="bazfoo", flaw=flaw)
+        pkg_c = PackageFactory(package="fobr", flaw=flaw)
+        PackageVerFactory(version="1.2.3.4", package=pkg_a)
+        PackageVerFactory(version="1.2.3.4", package=pkg_b)
+        PackageVerFactory(version="2.3.4.5", package=pkg_a)
+        PackageVerFactory(version="2.3.4.5", package=pkg_b)
+        PackageVerFactory(version="3.4.5.6", package=pkg_c)
+
+        fbqb = FlawBugzillaQueryBuilder(flaw)
+        fixed_in = fbqb.query.get("cf_fixed_in")
+
+        # Test that
+        # - Versions already in fixed_in are left in fixed_in as they are, including order and
+        #   dash/space separator.
+        # - Versions no longer existing are removed from fixed_in.
+        # - New versions are added to fixed_in with space separator.
```

Comment:
Expected behaviours should be in the docstring.

Reply:
Fixed in commit [49a61c3](https://github.com/RedHatProductSecurity/osidb/pull/307/commits/49a61c3bba8cac77e241879831f422a0b39de6d1)